packages/flutter/lib/src/widgets/page_storage.dart

```diff
@@ -129,7 +111,121 @@ class PageStorageBucket {
   }
 }
 
-/// A widget that establishes a page storage bucket for this widget subtree.
+/// Establish a subtree in which widgets can opt into persisting states after
+/// being destroyed.
+/// 
+/// [PageStorage] is used to save and restore values that can outlive the widget
+/// For example, when multiple pages are grouped in tabs, when a page is
+/// switched out, its widget is destroyed and its state is lost. By adding a
+/// [PageStorage] at the root and adding a [PageStorageKey] to each page, the
+/// page's state will be stored in its closest ancestor [PageStorage], and
+/// restored when it's switched back.
+/// Usually you don't need to explicitly use a [PageStorage], since it's already
+/// included in routes.
+/// [PageStorageKey] is used by [Scrollable] if
+/// `keepScrollOffset` is enabled to save their [ScrollPosition]s.
+/// {@tool snippet}
```

Comment:
Can you convert this snippet into using a template as well as dartpad, so that it's more concise to display and easier to read? 
Since your sample contains `bottomNavigationBar`, you'll probably have to use the freeform template, https://github.com/flutter/flutter/blob/master/dev/snippets/config/templates/freeform.tmpl

An example can be found at https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/inherited_theme.dart#L32